packages/dashboard/src/components/task-summary-panel.tsx

```diff
@@ -0,0 +1,221 @@
+import React from 'react';
+import { SpotlightValue } from './spotlight-value';
+import Debug from 'debug';
+import * as RomiCore from '@osrf/romi-js-core-interfaces';
+import { IconButton, makeStyles, Typography } from '@material-ui/core';
+import { TreeItem, TreeView } from '@material-ui/lab';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
+import { TaskSummaryPanelItem } from './task-summary-panel-item';
+import PlayCircleFilledWhiteIcon from '@material-ui/icons/PlayCircleFilledWhite';
+import TaskManager from '../managers/task-manager';
+import { colorPalette } from '../util/css-utils';
+import { TreeButtonGroup } from './tree-button-group';
+import { Theme } from '@material-ui/core/styles/createMuiTheme';
+
+const debug = Debug('OmniPanel:TaskSummaryPanel');
+
+export interface TaskSummaryPanelProps {
+  allTasks: RomiCore.TaskSummary[];
+  spotlight?: Readonly<SpotlightValue<string>>;
+}
+
+export const TaskSummaryPanel = React.memo((props: TaskSummaryPanelProps) => {
+  debug('task summary status panel render');
+
+  const { allTasks, spotlight } = props;
+  const classes = useStyles();
+
+  React.useEffect(() => {
+    if (!spotlight) {
+      return;
+    }
+    // TODO: spotlight
+  }, [spotlight]);
+
+  const [taskContents, setTaskContents] = React.useState<{
+    [key: string]: JSX.Element;
+  }>({});
+  const [expanded, setExpanded] = React.useState<string[]>([]);
+  const [selected, setSelected] = React.useState<string>('');
+
+  // We need to persist across the renders
+  const savedTasksContent = React.useRef<{
+    [key: string]: JSX.Element;
+  }>({});
+
+  const handleToggle = (event: React.ChangeEvent<{}>, nodeIds: string[]) => {
+    setExpanded(nodeIds);
+  };
+
+  const handleSelect = (event: React.ChangeEvent<{}>, nodeIds: string) => {
+    setSelected(nodeIds);
+  };
+
+  const handleResetTasks = () => {
+    setExpanded([]);
+    setSelected('');
+  };
+
+  const handleClearAllCurrTasks = () => {
```

Comment:
Why would the user want to clear current tasks? From what I understand, this only clears the task from the list, and does not actually cancel the tasks.

Reply:
Yeah, it only clears the task from the list and does not cancel the task.

I want to implement the task pause in a FUP PR. 

I just thought that they will want to clear it when they have a lot of old finished tasks on the list.

Reply:
Hmm, it's the lack of a backend database problem again. If we keep the previous tasks in memory it will be gone when we refresh the page, we can't persist the tasks without a db. I guess this is the best we can do now so that the user can see the result of the task. Can you add a TODO here?

Reply:
Sure thing!